apps/slippistats/slippistats.star

```diff
@@ -0,0 +1,277 @@
+    "26_3": "UklGRlYBAABXRUJQVlA4TEkBAAAvF8AFEH9gpo2UG58NC28Hj1gCGW/JSCQpSCJ8BiJQg97Ykx8BizQSScrJl+CoQQ0s/YcKyJfzHwQ6lTYJLlilSLEFLIE7/nHG75PbzAIUW9vWOG9c5f9nCmoY/OTPCpIyo6xjZgYdNSCzhE6XUM3Mjhn+86UkuZnuIaL/EwA355wjO9OECKvZ4kIXelhxsZBZqJtCdXFhNBYVirACIG4WGmZTY0R1icIV2VR47QEQFEXOJtGpB2BBrelr+4NOfAAPao3SoQz5wdrmwo03X87ClxfseFw3jiRlTr1ge/mFpU2bX3SWD/BeEWtY+rqnPID1an1vzfK+rwDgferW+05L36YCxqsLl3tuT2eWFXYxWFW4tyhPuk8UnuigwuHb8d7ptpx0wrIs656omyP9QESUqG55UpC2bfvxUZa2euGal3I90ccBxv8FAAA=",
+    "26_4": "UklGRlQBAABXRUJQVlA4TEgBAAAvF8AFEH9gpo2UG1/Cvw10sqIPCSDHaMAkvEEk+pJHW4mLvmGmto1X4W9mUUQgn2oy2BjnPxBYYqDhuDgEc6wHe685aqmBmqaNF0D5/3+NuqRvZBGHuD+elxa3F+5aSe7u9hT+O/SHS7sX1bXfkXjRbAl3l5Nf2BHR/wmAk4gIoaUqRIQVKk0kisQIxSHDzfREU1gOEkZ5RrqIUACkmemGWVEeZTlE+qavIv3DBSBeZAR2mZ9dgIxXK+zDU35aBCheLfcF2M3zkDWjEeWfdmDNXoC86k00Ln3sfl6AvJ1Mz67YtflxEqAWEVO6bv/wBCBb1NbjSt9P6xRArdbByX5V664CSUXpG41fz4MbivztL0y/PfI9NT0ptFjP6R1fvS1DNWH6IhHRL3MTQX9nZl4sqvIr0DVN83h8OdULcKx6vTuLrQRI+hcA",
+}
+
+def get_schema():
+    return schema.Schema(
+        version = "1",
+        fields = [
+            schema.Text(
+                id = "code",
+                name = "Slippi Code",
+                desc = "Ex: TRB#328",
+                icon = "user",
+            ),
+        ],
+    )
+
+def main(config):
+    totalTimeCached = cache.get("cache_totalTime")
+
+    if totalTimeCached != None:
+        print("Hit! Displaying cached data.")
+        totalGames = cache.get("cache_totalGames")
+        winrate = cache.get("cache_winrate")
+        iconName = cache.get("cache_iconName")
+
+        totalTime = totalTimeCached
+
+    else:
+        print("Miss! Calling SLP API.")
+        code = config.str("code", DEFAULT_CODE)
+        code = code.replace("#", "-")
+        GET_URL = CHART_SLP_SERVER_URL + code
+
+        rep = http.get(GET_URL)
+        if rep.status_code != 200:
+            fail("Chart SLP Request Failed with code")
+        totalTime = rep.json()["totalTime"]
+        totalTime = totalTime.split(":")
+        totalTime = humanize.comma(totalTime[0]) + " hrs"
+        totalGames = str(humanize.comma(int(rep.json()["totalMatches"]))) + "gs"
+        winrate = str(int(rep.json()["winrate"])) + "% wr"
+        iconName = str(int(rep.json()["main"])) + "_" + str(int(rep.json()["mainColor"]))
+
+        cache.set("cache_iconName", str(iconName), ttl_seconds = 43200)
+        cache.set("cache_totalGames", str(totalGames), ttl_seconds = 43200)
+        cache.set("cache_winrate", str(winrate), ttl_seconds = 43200)
+        cache.set("cache_totalTime", str(totalTime), ttl_seconds = 43200)  # this is like 12 hours
```

Comment:
I'd recommend something like

```
        data = {
            "totalTime": totalTime,
            "totalGames": totalGames,
            "winrate": winrate,
            "iconName": iconName,
        }

        cache.set(code, json.encode(data), ttl_seconds = 43200)
```

So you have only one entry to add/check in the cache and all the data's in one centralized object.

Reply:
Great idea -- I took it and implemented it. Thanks for all the assistance!!